File: frontend/src/types/car.ts
export type FuelType = "Any" | "Petrol" | "Diesel" | "Hybrid" | "Electric";

export type PriorityKey = "price" | "safety" | "mileage" | "features" | "space";

export interface RecommendationFilters {
  budget_min: number;
  budget_max: number;
  seats: number | null; // null = "Any"
  fuel: FuelType;
  min_safety_rating: number | null; // null = "Any"
  min_mileage: number;
  features: string[];
  priorities: PriorityKey[]; // max 2
}

export interface ScoreBreakdown {
  budget?: number;
  safety?: number;
  mileage?: number;
  features?: number;
  space?: number;
  [key: string]: number | undefined;
}

export interface CarDetails {
  id: string;
  make: string;
  model: string;
  variant: string;
  min_price: number;
  max_price: number;
  body_type: string;
  fuel: string;
  seats: number;
  mileage: number;
  safety_rating: number;
  features: string[];
  score?: number; // present on some recommendation/detail variants
  score_breakdown?: ScoreBreakdown;
  highlights?: string[]; // short explanation bullets
  image_url?: string;
  description?: string;
  engine?: string;
  transmission?: string;
  boot_space_litres?: number;
  ground_clearance_mm?: number;
  airbags?: number;
  warranty?: string;
}

export interface RecommendationResult {
  score: number; // overall match score, 0-100
  score_breakdown?: ScoreBreakdown;
  cars: CarDetails[];
}

export interface RecommendationResponse {
  results: RecommendationResult[];
  count: number;
}

export type SortOption = "best_match" | "lowest_price" | "highest_safety" | "best_mileage";

File: backend/internals/models/car.go
package models

import (
	"encoding/json"
	"os"

	"github.com/google/uuid"
)

type Car struct {
	ID           uuid.UUID `json:"id"`
	Make         string    `json:"make"`
	Model        string    `json:"model"`
	Variant      string    `json:"variant"`
	MinPrice     float64   `json:"min_price"`
	MaxPrice     float64   `json:"max_price"`
	BodyType     string    `json:"body_type"`
	Fuel         string    `json:"fuel"`
	Seats        int       `json:"seats"`
	Mileage      float64   `json:"mileage"`
	SafetyRating float64   `json:"safety_rating"`
	Features     []string  `json:"features"`
}

func LoadCars(path string) ([]Car, error) {
	file, err := os.ReadFile(path)
	if err != nil {
		return nil, err
	}

	var cars []Car
	if err := json.Unmarshal(file, &cars); err != nil {
		return nil, err
	}

	return cars, nil
}
